### ISSUE TITLE
broken promise // hung system // broken hearts

### ISSUE THREAD

**Issue by dalehenrich** [GemTalk/JadeiteForPharo#70]
Using a recently built JfP:
```
JadeiteForPharo 82dc6f0
PharoGemStoneFFI 3575de4
RemoteServiceReplication f90b9c1
---------------------------------------------
RowanClientServices d78cc35
```
And  my development image (connection details available upon request) the following sequence appears results in unfulfilled RSR promises:

1. set breakpoint in method (break method RwPrjLoadToolV2 _loadProjectSetDefinition:instanceMigrator:symbolList: @9)
2. using SUnit browser debug RwBrowserTool48991Test>>testMoveClass_old ... which should hit breakpoint
3. hung Jadeite...

At logout, I get a couple of walkbacks for broken promises ...

**Comment by dalehenrich** [comment]
Stashed a snapshot of the 3.7.1 extent I was using here: /bosch1/users/dhenrich/_stones/37x/stones/rowan3_dev_h/snapshots/extent0.issue_70.dbf and $GEMSTONE is /bosch1/users/dhenrich/_stones/37x/h_37x/noop50/gs/product ... a private build

**Comment by ericwinger** [comment]
Can you try again with the latest RowanClientServices from ericV3.0_pharo branch loaded? Looks like you're version is a bit behind. 

5a7a68c 2 days ago update class after add or rename category
26b82ef 4 days ago make sure previous commit committed
f37572f 4 days ago file write - truncate first
03e4433 7 days ago better error handling after disconnection
c872edd 2 weeks ago update added projects before return to client
dad4741 2 weeks ago check for added projects
9a19e62 2 weeks ago took out reload after checkout
bb78ac8 2 weeks ago work in progress - post checkout testing
76e39ec 2 weeks ago work in progress #52 - git checkout options
8a67256 2 weeks ago Fixed #18 - setting break fails + remove super/sub calc + other  fixes
45db83b 3 weeks ago Merge branch 'dkhV3.0_pharo' into ericV3.0_pharo
d78cc35 4 weeks ago patch to allow use of debugger in both Dolphin and Pharo ... thanks to Eric

**Comment by dalehenrich** [comment]
confirm that running with latest RowanClientServices the bug DOES NOT REPRODUCE!
```
JadeiteForPharo 82dc6f0
PharoGemStoneFFI 3575de4
RemoteServiceReplication f90b9c1
---------------------------------------------
RowanClientServices 5a7a68c
```
Continuing past breakpoints allows the test to complete with no hangs:
![image](https://github.com/GemTalk/JadeiteForPharo/assets/623951/5b99268c-4a68-4bc5-875e-91c2466313fe)
